src/main/java/teammates/common/datatransfer/FeedbackRubricQuestionDetails.java

```diff
@@ -52,17 +55,26 @@ public boolean extractQuestionDetails(
         int numOfRubricChoices = Integer.parseInt(numOfRubricChoicesString);
         int numOfRubricSubQuestions = Integer.parseInt(numOfRubricSubQuestionsString);
         List<String> rubricChoices = new ArrayList<String>();
+        List<Integer> rubricWeights = new ArrayList<Integer>();
         List<String> rubricSubQuestions = new ArrayList<String>();
         List<List<String>> rubricDescriptions = new ArrayList<List<String>>();
         
         int numActualChoices = 0;
         int numActualSubQuestions = 0;
         
-        // Get list of choices
+        // Get list of choices and weights
         for(int i = 0 ; i<numOfRubricChoices ; i++) {
             String choice = HttpRequestHelper.getValueFromParamMap(requestParameters, Const.ParamsNames.FEEDBACK_QUESTION_RUBRIC_CHOICE + "-" + i);
-            if(choice != null) {
+            String weight = HttpRequestHelper.getValueFromParamMap(requestParameters, Const.ParamsNames.FEEDBACK_QUESTION_RUBRIC_WEIGHT + "-" + i);
+            if(choice != null && weight != null) {
                 rubricChoices.add(choice);
+
+                try {
+                    rubricWeights.add(Integer.parseInt(weight));
+                } catch (NumberFormatException e) {
+                    rubricWeights.add(0);
```

Comment:
Right, I forgot <code>validateQuestionDetails</code> is called only after the question details have been constructed. I'm not very sure what to feel about storing <code>null</code> as a value in the list.